app/javascript/src/components/Dashboard/Contacts/ContactTable.jsx

```diff
@@ -0,0 +1,150 @@
+import React, { useState, useEffect } from "react";
+import { Checkbox, Button, Tooltip, PageLoader, Avatar } from "neetoui";
+import contactsApi from "apis/contacts";
+import DeleteAlert from "../../Common/DeleteAlert";
+
+export default function ContactTable({
+  selectedContactIds,
+  setSelectedContactIds,
+  setContacts,
+  contacts = [],
+}) {
+  const [showDeleteAlert, setShowDeleteAlert] = useState(false);
+  const [loading, setLoading] = useState(true);
+  const [deletingContact, setDeletingContact] = useState("");
+
+  useEffect(() => {
+    fetchContacts();
+  }, []);
+
+  const fetchContacts = async () => {
+    try {
+      setLoading(true);
+      const response = await contactsApi.fetch();
+      setContacts(response.data);
+    } catch (error) {
+      logger.error(error);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const setDeletingRecord = noteId => {
+    setShowDeleteAlert(true);
+    setDeletingContact(noteId);
+  };
+
+  if (loading) {
+    return <PageLoader />;
+  }
+
+  return (
+    <div className="w-full px-4">
+      <table className="nui-table nui-table--checkbox">
+        <thead>
+          <tr>
+            <th>
+              <Checkbox
+                checked={
+                  selectedContactIds.length ===
+                  contacts.map(contact => contact.id).length
+                }
+                onClick={() => {
+                  const contactIds = contacts.map(contact => contact.id);
+                  if (selectedContactIds.length === contactIds.length) {
+                    setSelectedContactIds([]);
+                  } else {
+                    setSelectedContactIds(contactIds);
+                  }
+                }}
+              />
+            </th>
+            <th className="text-left">Name</th>
+            <th className="text-left">Email</th>
+            <th className="text-center">Department</th>
+            <th className="text-center">Contact Number</th>
+            <th className="text-center">Add to Basecamp</th>
```

Comment:
Please create an empty `<th></th>` here. 
This will be corresponding to the two IconButtons that we have from Delete and Edit actions.